modules/instrument_builder/js/instrument_builder.instrument.js

```diff
@@ -297,10 +297,14 @@ var Instrument = {
                             break;
                     }
 
+                    /* "Header" elements always have 'undefined' as their Name. These should not trigger an error message. */
                     if (elementNames.indexOf(tempElement.Name) < 0) {
-                      elementNames.push(tempElement.Name)
+                      if (tempElement.Name != "undefined" && tempElement.Type != "header") {
```

Comment:
You don't need to check for type here.  
The easier/cleaner solution is to wrap the code into an `if`
```js
if (tempElement.Name) {
   if (elementNames.indexOf(tempElement.Name) < 0) {
    ....
   } else {
    ....
   }
}
```

PS: Note that `undefined` is a type not a string. It still works because you're doing loose equality `!= "undefined"` which is not a good practice. [See explanation](https://www.impressivewebs.com/why-use-triple-equals-javascipt/)